### PR TITLE
Practice for chapter 7

### PR DESCRIPTION
**Pull request by VuThiTranVan**
Sign up

### CODE REVIEW

**Comment by VuThiTranVan** [comment]
screenshot from 2017-10-09 15-47-24

**Comment by VuThiTranVan** [comment]
ready

File: sample_app/app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,3 +1,34 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find_by(id: params[:id])
```
Comment (dothidiemthao) [inline sample_app/app/controllers/users_controller.rb]: bỏ ( )

File: sample_app/app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+    flash[:danger] = t "users.show.no_user_msg"
+    redirect_to signup_path
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (dothidiemthao) [inline sample_app/app/controllers/users_controller.rb]: bỏ ( )

File: sample_app/app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<% provide(:title, @user.name) %>
```
Comment (dothidiemthao) [inline sample_app/app/views/users/show.html.erb]: bỏ ( )

File: sample_app/app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,3 +1,34 @@
...
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user)
+      .permit(
```
Comment (dothidiemthao) [inline sample_app/app/controllers/users_controller.rb]: bỏ ( )

**Comment by dothidiemthao** [comment]
commented

**Comment by VuThiTranVan** [comment]
screenshot from 2017-10-09 16-12-57

**Comment by VuThiTranVan** [comment]
ready

**Review by dothidiemthao — APPROVED** [review]

**Pull request merged**